pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -532,6 +532,10 @@ func (r *Reconciler) reconcile(
 
 					patch := client.MergeFrom(garden.DeepCopy())
 					garden.Status.EncryptedResources = shared.StringifyGroupResources(encryptedResources)
+					if garden.Status.Credentials == nil {
```

Comment:
Same as above, can we avoid creating this field if `encryptedResources` is empty?